libcontainer/nsenter/nsenter.go

```diff
@@ -3,7 +3,7 @@
 package nsenter
 
 /*
-#cgo CFLAGS: -Wall
+#cgo CFLAGS: -Wall -Werror
```

Comment:
Given that what constitutes a "warning" varies from version to version of the compiler, shouldn't this be somehow added only in the CI instead? Otherwise I imagine a lot of downstream packagers might end up having to patch it out in the future. 😬

Reply:
Totally agree with @tianon -- it is discouraged to have `-Werror` in non-development builds, for the reason explained above.

It seems we can have `CGO_CFLAGS=-Werror` defined via environment in CI (and CI only). Maybe also add `-Wextra` (to `nsenter.go`, not via env, as otherwise runtime/cgo will fail to build, as it has `-Werror` -- and that's a nice demonstration of why -Werror can be bad).

Reply:
No, we can't have `-Wextra`:

```console
# github.com/opencontainers/runc/libcontainer/nsenter
cloned_binary.c: In function ‘clone_binary’:
cloned_binary.c:511:21: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘__off_t’ {aka ‘long int’} [-Wsign-compare]
  511 |         while (sent < statbuf.st_size) {
      |                     ^
cloned_binary.c:522:18: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘__off_t’ {aka ‘long int’} [-Wsign-compare]
  522 |         if (sent != statbuf.st_size)
      |                  ^~
# github.com/opencontainers/runc/libcontainer/nsenter
nsexec.c: In function ‘write_file’:
nsexec.c:202:17: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
  202 |         if (len != data_len) {
      |                 ^~
nsexec.c: In function ‘try_mapping_tool’:
nsexec.c:246:73: warning: unused parameter ‘map_len’ [-Wunused-parameter]
  246 | static int try_mapping_tool(const char *app, int pid, char *map, size_t map_len)
      |                                                                  ~~~~~~~^~~~~~~
# github.com/opencontainers/runc/libcontainer/nsenter
_cgo_main.c: In function ‘crosscall2’:
_cgo_main.c:2:23: warning: unused parameter ‘fn’ [-Wunused-parameter]
    2 | void crosscall2(void(*fn)(void*), void *a, int c, __SIZE_TYPE__ ctxt) { }
      |                 ~~~~~~^~~~~~~~~~
_cgo_main.c:2:41: warning: unused parameter ‘a’ [-Wunused-parameter]
    2 | void crosscall2(void(*fn)(void*), void *a, int c, __SIZE_TYPE__ ctxt) { }
      |                                   ~~~~~~^
_cgo_main.c:2:48: warning: unused parameter ‘c’ [-Wunused-parameter]
    2 | void crosscall2(void(*fn)(void*), void *a, int c, __SIZE_TYPE__ ctxt) { }
      |                                            ~~~~^
_cgo_main.c:2:65: warning: unused parameter ‘ctxt’ [-Wunused-parameter]
    2 | void crosscall2(void(*fn)(void*), void *a, int c, __SIZE_TYPE__ ctxt) { }
      |                                                                 ^
_cgo_main.c: In function ‘_cgo_release_context’:
_cgo_main.c:4:41: warning: unused parameter ‘ctxt’ [-Wunused-parameter]
    4 | void _cgo_release_context(__SIZE_TYPE__ ctxt) { }
      |                                         ^
_cgo_main.c: In function ‘_cgo_allocate’:
_cgo_main.c:6:26: warning: unused parameter ‘a’ [-Wunused-parameter]
    6 | void _cgo_allocate(void *a, int c) { }
      |                    ~~~~~~^
_cgo_main.c:6:33: warning: unused parameter ‘c’ [-Wunused-parameter]
    6 | void _cgo_allocate(void *a, int c) { }
      |                             ~~~~^
_cgo_main.c: In function ‘_cgo_panic’:
_cgo_main.c:7:23: warning: unused parameter ‘a’ [-Wunused-parameter]
    7 | void _cgo_panic(void *a, int c) { }
      |                 ~~~~~~^
_cgo_main.c:7:30: warning: unused parameter ‘c’ [-Wunused-parameter]
    7 | void _cgo_panic(void *a, int c) { }
      |                          ~~~~^
```

Reply:
@tianon @kolyshkin oh, right. Excellent point!